test/integration/deprecated_specs.cc

```diff
@@ -38,3 +42,89 @@ TEST(DeprecatedSpecs, Spec1_2)
   sdf::SDFPtr sdf(new sdf::SDF());
   EXPECT_FALSE(sdf::initFile(filename, sdf));
 }
+
+////////////////////////////////////////////////////
+TEST(DeprecatedElements, CanEmitErrors)
+{
+  sdf::SDFPtr sdf(new sdf::SDF());
+  sdf::init(sdf);
+  auto elem = std::make_shared<sdf::Element>();
+  elem->SetRequired("-1");
+  elem->SetName("testElem");
+  // Change the root of sdf so we can test "testElem" in isolation
+  sdf->Root(elem);
+  auto config = sdf::ParserConfig::GlobalConfig();
+  config.SetWarningsPolicy(sdf::EnforcementPolicy::ERR);
+  sdf::Errors errors;
+  EXPECT_TRUE(sdf::readString(
+      "<sdf version='1.8'><testElem/></sdf>", config, sdf, errors));
+  ASSERT_FALSE(errors.empty());
+  EXPECT_EQ(sdf::ErrorCode::ELEMENT_DEPRECATED, errors[0].Code());
+}
+
+////////////////////////////////////////////////////
+TEST(DeprecatedElements, CanEmitWarningWithErrorEnforcmentPolicy)
+{
+  // Capture sdfwarn output, which is treamed to std::cerr
```

Comment:
nit Also ew... it seems odd that the only way to intercept logging is via C++ I/O redirect on `cerr`.

Can't `Console` admit a more intuitive redirect as public API? (via `msgStream` or `logFileStream`?)
(lemme know if issue is better here)

Reply:
Added a way redirect `Console` in https://github.com/osrf/sdformat/pull/543/commits/c6b9843f2bf067ed404c720e7130040fb945ee79. Let me know if the API makes sense.

Reply:
Yup, awesome!